mm2src/mm2_main/src/lp_swap/swap_v2_common.rs

```diff
@@ -371,3 +373,99 @@ pub(super) async fn swap_kickstart_handler<
         error!("Error {} on trying to run the swap {}", e, uuid);
     }
 }
+
+pub(super) async fn swap_kickstart_handler_for_maker(
+    ctx: MmArc,
+    swap_repr: MakerSwapDbRepr,
+    storage: MakerSwapStorage,
+    uuid: Uuid,
+) {
+    let (maker_coin, taker_coin) = match swap_kickstart_coins(&ctx, &swap_repr, &uuid).await {
+        Some(coins) => coins,
+        None => return,
+    };
+
+    match (maker_coin, taker_coin) {
+        (MmCoinEnum::UtxoCoin(m), MmCoinEnum::UtxoCoin(t)) => {
+            swap_kickstart_handler::<
+                MakerSwapStateMachine<UtxoStandardCoin, UtxoStandardCoin>,
+                UtxoStandardCoin,
+                UtxoStandardCoin,
+            >(swap_repr, storage, uuid, m, t)
+            .await
+        },
+        (MmCoinEnum::EthCoin(m), MmCoinEnum::EthCoin(t)) => {
+            swap_kickstart_handler::<MakerSwapStateMachine<EthCoin, EthCoin>, EthCoin, EthCoin>(
+                swap_repr, storage, uuid, m, t,
+            )
+            .await
+        },
+        (MmCoinEnum::UtxoCoin(m), MmCoinEnum::EthCoin(t)) => {
+            swap_kickstart_handler::<MakerSwapStateMachine<UtxoStandardCoin, EthCoin>, UtxoStandardCoin, EthCoin>(
```

Comment:
looks like there is some redundancy in these type args. would be nice if we can remove that.